examples/ppm_example.py

```diff
@@ -0,0 +1,43 @@
+from pylab import *
+from pyspecdata import *
+
+# make peaks on a frequency axis
+x = nddata(r_[0:8:49152j], "t2")
+x.set_units("t2", "s")
+
+# generate time-domain signal
+signal = 0 * x  # create an array of zeros that's the right shape
+SFO1 = 400.130438262389
+TMS_shift = 12.5
+for A, nu, R in [
+    (0.1, TMS_shift, 0.04),
+    (1, 25, 0.08),
+    (1, 50, 1.2),
+    (1, 20, 0.45),
+    (1, 750, 1.2),
+    (1, 10, 0.08),
+]:
+    nu = SFO1 + nu # it's really unclear why this is done!
+    signal += A * exp(1j * 2 * pi * nu * x - x / R)
+signal.set_units("t2", "s")
+signal.ft("t2", shift=True)
+
+# Parameters needed by to_ppm to work
+signal.set_prop('acq',{'SFO1':400.130438262389,'SW_h':4807.69230769231,'O1':438.262389})
+signal.set_prop('proc',{'OFFSET':12.50096})
```

Comment:
the manual implies that this is `(SFO1/SF-1) * 1.0e6 + 0.5 * SW * SFO1/SF` for some acquisition modes, and without the second term for other acquisition modes -- what is SF, and is that true here? (I think SF is BF1, and so would depend on the dataset where these parameters are coming from)

Reply:
If SF is BF1, then the first term is equivalent to `(SFO1-BF1)*1e6/BF1=O1/BF1` so that would be the offset from BF1 in ppm.  The second term I think is supposed to be half SW in units of ppm --> I think we actually see the difference in "offset" for the two modes if you look at the Hz axis for 2H vs. 1H -- the former has 0 on one side and for the other it's centered.

Reply:
ahh -- see [here](http://www2.chem.uic.edu/nmr/downloads/processing-reference.pdf) pg 259 (search "basically makes BF1 = SF") which seems to imply that SF is related to BF1 using the parameter SR

Reply:
aha!  This is related to the above question --> if you look up the definition of `SR`, it looks like `SF=SR+BF1` and it defines `SF` as the absolute frequency of the TMS.

see [p35 + p37 here](https://www.pascal-man.com/pulseprogram/avance3/topspin_2_1/proc_commands_parameters.pdf) and compare to our manual please.

The remaining question for our purposes is then which of these (`OFFSET`, `BF`, `SR`, `SF`) is actually stored, so that we can include the TMS referencing.

Reply:
BF is an acquisition parameter and is stored.
OFFSET, SR, and SF are all stored in processing parameters.